src/places_nearby.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright  (C) 2020, Kisio Digital and/or its affiliates. All rights reserved.
+//
+// This file is part of Navitia,
+// the software to build cool stuff with public transport.
+//
+// Hope you'll enjoy and contribute to this project,
+// powered by Kisio Digital (www.kisio.com).
+// Help us simplify mobility and open public transport:
+// a non ending quest to the responsive locomotion way of traveling!
+//
+// This contribution is a part of the research and development work of the
+// IVA Project which aims to enhance traveler information and is carried out
+// under the leadership of the Technological Research Institute SystemX,
+// with the partnership and support of the transport organization authority
+// Ile-De-France Mobilités (IDFM), SNCF, and public funds
+// under the scope of the French Program "Investissements d’Avenir".
+//
+// LICENCE: This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+// Stay tuned using
+// twitter @navitia
+// channel `#navitia` on riot https://riot.im/app/#/room/#navitia:matrix.org
+// https://groups.google.com/d/forum/navitia
+// www.navitia.io
+
+extern crate static_assertions;
+
+use super::geometry::{bounding_box, distance_coord_to_coord};
+use crate::models::{ModelRefs, StopPointIdx};
+use regex::Regex;
+use std::fmt::{Display, Formatter};
+use transit_model::objects::{Coord, StopPoint};
+use typed_index_collection::Iter;
+
+pub fn places_nearby_impl<'model, 'uri>(
+    model: &'model ModelRefs,
+    uri: &'uri str,
+    radius: f64,
+) -> Result<PlacesNearbyResult<'model>, BadPlacesNearby> {
+    // ep: entrypoint
+    let ep_coord = parse_entrypoint(model, uri)?;
+    let bounding_box = bounding_box(ep_coord, radius);
+
+    Ok(PlacesNearbyResult::new(
+        model,
+        ep_coord,
+        radius,
+        bounding_box,
+    ))
+}
+
+#[derive(Debug)]
+pub struct PlacesNearbyResult<'model> {
```

Comment:
```suggestion
pub struct PlacesNearbyIter<'model> {
```
Seeing the name Result, I thought it was something like Result<Ok, Err>.